tasks/configure.yml

```diff
@@ -1,6 +1,21 @@
 ---
 
-- name: Ensure Docker service is restarted
+- name: Add thinpool configuration
+  become: yes
+  template:
+    dest: /etc/docker/daemon.json
+    group: root
+    mode: 0444
+    owner: root
+    src: daemon.json.j2
+  when: docker.thin_pool_device is not none
+  notify:
+    - restart docker
+
+- name: Flush handlers to ensure everything is up to date
+  meta: flush_handlers
```

Comment:
Do you need an immediate restart here or can it just wait for the handler to kick in later on?

Reply:
It is required at this point, rather than at the very end of the ansible run. The restart of docker mounts the thin pool device, which removes any images/containers that were loaded into /var/lib/docker before the mount. 
As soon as this sansible role finishes, I am loading the aws agent image into /var/lib/docker, and then starting a container based on this image. If that happens pre-restart, the container is nuked.

Reply:
OK cool, in that case let's a more targeted version like so:

```YAML
- name: Add thinpool configuration
  become: yes
  template:
    dest: /etc/docker/daemon.json
    group: root
    mode: 0444
    owner: root
    src: daemon.json.j2
  when: docker.thin_pool_device is not none
  register: docker_thinpool_config

- name: Ensure Docker service is restarted in thinpool change
  become: yes
  service:
    name: docker
    state: restarted
  when: docker_thinpool_config.changed

- name: Ensure Docker service starts at boot
  become: yes
  service:
    name: docker
    state: started
```

Reply:
ok. :tada: